File: RAI/Clase15/Python/InnerClasses/Herramientas.py
#Outer Class
class Herramientas:

    #Inner Class
    class Martillo:
        def __init__(self, tamano:str, empunadura: str, mazo:str):
            self.tamano = tamano
            self.empunadura = empunadura
            self.mazo = mazo
        
        def __str__(self) -> str:
            return "Martillo de " + str(self.tamano) + " onzas, empunadura de " + self.empunadura + " y mazo de " + self.mazo

    def __init__(self, nombre: str):
        self.nombre = nombre
        self.martillo = self.Martillo(32, "madera", "acero")

File: RAI/Clase15/Java/InnerClasses/Herramientas.java
//package RAI.Clase15.Java.InnerClasses;

//Outer Class
public class Herramientas {
    private String nombre;
    private Martillo martillo;
    
    //Inner Class
    public class Martillo{
        private int tamano;
        private String empunadura;
        private String mazo;

        public Martillo(int tamano, String empunadura, String mazo){
            this.tamano = tamano;
            this.empunadura = empunadura;
            this.mazo = mazo;
        }

        @Override
        public String toString(){
            return "Martillo de " + this.tamano + " onzas, empunadura de " + this.empunadura + " y mazo de " + this.mazo;
        }
    }

    public Herramientas(String nombre){
        this.nombre = nombre;
        this.martillo = new Martillo(32, "madera", "Acero");
    }

    public Martillo getMartillo(){
        return this.martillo;
    }

    public Martillo crearMartillo(int tamano, String empunadura, String mazo){
        return new Martillo(tamano, empunadura, mazo);
    }
}
